File: src/main/scala/de/theunknownxy/jgui/base/Point.scala
package de.theunknownxy.jgui.base

class Point(var x: Float, var y: Float) {
  def <(that: Point) : Boolean = this.x < that.x && this.y < that.y
  def <=(that: Point) : Boolean = this.x <= that.x && this.y <= that.y
  def >(that: Point) : Boolean = this.x > that.x && this.y > that.y
  def >=(that: Point) : Boolean = this.x >= that.x && this.y >= that.y

  override def equals(o: Any) : Boolean = {
    o match {
      case that: Point => x == that.x && y == that.y
      case _ => false
    }
  }
}

File: src/main/kotlin/de/theunknownxy/mcdocs/gui/base/Point.kt
package de.theunknownxy.mcdocs.gui.base

/**
 * Created by marco on 2/28/15.
 */
public class Point(var x: Float, var y: Float) {
    override fun equals(other: Any?): Boolean {
        if (other != null) {
            if (other is Point) {
                return other.x == x && other.y == y
            }
        }
        return false
    }
}